lib/schema.ts

```diff
@@ -164,19 +171,21 @@ function buildFields(schema, rLevelParentMax, dLevelParentMax, path) {
   return fieldList;
 }
 
-function listFields(fields) {
-  let list = [];
+function listFields(fields: Record<string, ParquetField>) {
+  let list: Array<ParquetField> = [];
 
   for (let k in fields) {
     list.push(fields[k]);
 
-    if (fields[k].isNested) {
-      list = list.concat(listFields(fields[k].fields));
+    let field = fields[k].fields
```

Comment:
Plural and I don't think this needs to be a `let`...
```suggestion
    const fields = fields[k].fields
```